lib/kubernetes-deploy/kubectl.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+require 'kubeclient'
+
+module KubernetesDeploy
+  module Kubectl
+    class KubectlError < FatalDeploymentError; end
+
+    def run_kubectl(*args, namespace: "", context: "", namespaced: true, with_context: true)
```

Comment:
From the API part, it seems weird that you have to pass both `namespace` and `namespaced`. If you don't want it to be namespaced, just skip passing `namespace`.